pkg/mpc-examples/contracts/00-null-controller/NullControllerFactory.sol

```diff
@@ -0,0 +1,89 @@
+
+/**
+ * @title NullControllerFactory
+ * @notice Factory for a Managed Pool and NullController.
+ * @dev Determines controller deployment address, deploys pool (w/ controller address as argument), then controller.
+ */
+contract NullControllerFactory {
+    mapping(address => bool) public isControllerFromFactory;
+
+    address public managedPoolFactory;
+    IVault public balancerVault;
+    bool public isDisabled;
+
+    uint256 private _nextControllerSalt;
+    address private _lastCreatedPool;
+
+    event ControllerCreated(address indexed controller, IVault vault, bytes32 poolId);
+
+    constructor(IVault vault, address factory) {
+        balancerVault = vault;
+        managedPoolFactory = factory;
+    }
+
+    /**
+     * @dev Return the address of the most recently created pool.
+     */
+    function getLastCreatedPool() external view returns (address) {
+        return _lastCreatedPool;
+    }
+
+    function create(IManagedPoolFactory.NewPoolParams memory params) external {
+        if (!isDisabled) {
+            bytes32 controllerSalt = bytes32(_nextControllerSalt);
+            _nextControllerSalt += 1;
+
+            bytes memory controllerCreationCode = abi.encodePacked(
+                type(NullController).creationCode,
+                abi.encode(balancerVault)
+            );
+            address expectedControllerAddress = Create2.computeAddress(
+                controllerSalt,
+                keccak256(controllerCreationCode)
+            );
+
+            // build arguments to deploy pool from factory
+            address[] memory assetManagers = new address[](params.tokens.length);
+            for (uint256 i = 0; i < assetManagers.length; i++) {
+                assetManagers[i] = expectedControllerAddress;
+            }
+
+            // TODO: instead of accepting the entirety of IManagedPoolFactory.NewPoolParams and
+            //       overwriting assetManagers and mustAllowlistLPs should we instead:
+            //          * extract all arguments and pass them individually?
+            //          * make a separate struct of the non-overwritten args, and use that to populate?
+            params.assetManagers = assetManagers;
+            params.mustAllowlistLPs = false;
+
+            IManagedPool pool = IManagedPool(
+                IManagedPoolFactory(managedPoolFactory).create(params, expectedControllerAddress)
+            );
+            _lastCreatedPool = address(pool);
+
+            address actualControllerAddress = Create2.deploy(0, controllerSalt, controllerCreationCode);
+            require(expectedControllerAddress == actualControllerAddress, "Deploy failed");
+
+            // log controller locally
+            isControllerFromFactory[actualControllerAddress] = true;
+
+            // log controller publicly
+            emit ControllerCreated(actualControllerAddress, balancerVault, pool.getPoolId());
+        }
+    }
+
+    // TODO: access control
```

Comment:
Let's include a solution within this PR. How about `SingletonAuthentication`, like the `BasePoolFactory` [uses](https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-utils/contracts/factories/BasePoolFactory.sol#L62)?

Reply:
I think SingletonAuthentication could potentially be a solution for real, production controllers, but feels like it might go against the ethos of what (I imagined) this repo is supposed to be.

While it is a well-built solution, it adds a good deal of complexity as far as understanding what's happening under the hood. This would require a prospective controller designer to understand the requirements from governance, would force them to think about `actionId`s, and could potentially result in someone else deciding to brick their factory.

I don't know what the broad solution should be for controllers, but for this specific case, I think something like making the factory `Ownable` and having `disable` being `onlyOwner` would be simple, clear, and sufficient.

Reply:
You're absolutely right about this.

That makes me wonder, does it need to have a `disable()` function at all? Do you think that's useful/necessary to demonstrate in this repo?

If you think it is useful, then I agree with making this factory `Ownable`.

Reply:
done

kept disable, made Ownable/onlyOwner